sklearn/semi_supervised/tests/test_self_training.py

```diff
@@ -343,3 +344,111 @@ def test_self_training_estimator_attribute_error():
         self_training.fit(X_train, y_train_missing_labels).decision_function(X_train)
     assert isinstance(exec_info.value.__cause__, AttributeError)
     assert inner_msg in str(exec_info.value.__cause__)
+
+
+# Test that metadata is routed correctly for pipelines
+# ====================================================
+
+
+class SimpleEstimator(BaseEstimator):
+    classes_ = [0, 1, 2]
+
+    # This class is used in this section for testing routing in the pipeline.
+    # This class should have every set_{method}_request
```

Comment:
since this is a copy/paste, I think we should move it to the common file.

Reply:
Should I move it for Pipeline as well? I did this in commit 

But lmk if you prefer that I revert.

Reply:
Yes, moving for pipeline makes sense as well.